contracts/Quest1155.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.18;
+
+import {IERC1155} from '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
+import {ERC1155Holder} from '@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol';
+import {PausableUpgradeable} from '@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol';
+import {ReentrancyGuardUpgradeable} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {Ownable} from 'solady/src/auth/Ownable.sol';
+import {SafeTransferLib} from 'solady/src/utils/SafeTransferLib.sol';
+import {QuestFactory} from './QuestFactory.sol';
+
+/// @title Quest
+/// @author RabbitHole.gg
+/// @notice This contract is the Erc1155Quest contract. It is a quest that is redeemable for ERC1155 tokens.
+/// @dev This contract will not work with RabbitHoleReceipt
+contract Quest1155 is ERC1155Holder, ReentrancyGuardUpgradeable, PausableUpgradeable, Ownable {
+    using SafeTransferLib for address;
+
+    QuestFactory public questFactoryContract;
+    bool public queued;
+    address public protocolFeeRecipient;
+    address public rewardToken;
+    uint public endTime;
+    uint public startTime;
+    uint public totalParticipants;
+    uint public tokenId;
+    uint public redeemedTokens;
+    uint public questFee;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    error EndTimeInPast();
+    error EndTimeLessThanOrEqualToStartTime();
+    error InsufficientTokenBalance();
+    error InsufficientETHBalance();
+    error NotStarted();
+    error NotEnded();
+    error NotQueued();
+    error NotQuestFactory();
+    error QuestEnded();
+
+    event ClaimedSingle(address indexed account, address rewardAddress, uint amount);
+    event Queued(uint timestamp);
+
+    function initialize(
+        address rewardTokenAddress_,
+        uint endTime_,
+        uint startTime_,
+        uint totalParticipants_,
+        uint tokenId_,
+        uint questFee_,
```

Comment:
Wondering if it makes sense to pass in `questFee`. Seems like this contract doesn't utilize it (plus it can deviate from the `nftQuestFee` in the QuestFactory